varnish/src/vsc_wrapper.rs

```diff
@@ -0,0 +1,65 @@
+use std::ops::{Deref, DerefMut};
+use std::ffi::CString;
+use std::mem::size_of;
+use varnish_sys::ffi::{vsc_seg, VRT_VSC_Alloc, VRT_VSC_Destroy};
+
+pub unsafe trait VscMetric {
+    fn get_metadata() -> &'static str;
+}
+
+pub struct Vsc<T: VscMetric> {
+    metric: *mut T,
+    seg: *mut vsc_seg,
+    name: CString,
+}
+
+impl<T: VscMetric> Vsc<T> {
+    pub fn new(module_name: &str, module_prefix: &str) -> Self {
```

Comment:
Do we need any kind of context pointer here?  Or does `VRT_VSC_Alloc` rely on a global singleton?

Reply:
singleton (`heritage` in the C code)